### ISSUE TITLE
Session TTL is used for assume role duration on SAML-assumed roles

### ISSUE THREAD

**Issue by devonbleak** [segmentio/aws-okta#105]
What this effectively means is that this construct ignores the assume_role_ttl and other ways of setting that value:
```
# example with a role that's configured with a max session duration of 12 hours
[profile ttldemo]
aws_saml_url = home/amazon_aws/cuZGoka9dAIFcyG0UllG/214
role_arn = arn:aws:iam::<account-id>:role/<okta-role-name>
assume_role_ttl = 12h
```

It instead uses the session ttl setting.

IMO we should be checking for source profile = target profile and in that case leverage assume_role_ttl and return fresh credentials on each call rather than the previously cached creds so that the expiration is more consistent.  We should continue to operate as we are currently when source profile != target profile.

**Comment by nickatsegment** [comment]
Not sure if this is still the case since @Fauzyy did some work there recently.

**Comment by Fauzyy** [comment]
I feel like we should be using the session ttl in the case where source profile = target profile or when there is no source profile. It looks like the assume role ttl was originally intended for AWS assumed roles (and not SAML assumed roles).

It's still confusing though, do you see any benefit to keeping these separate? It would be nice to merge them into one `role_ttl` option.

@devonbleak let me know your thoughts.

**Comment by devonbleak** [comment]
@Fauzyy @nickatsegment I looked at the updated code and from what I can tell the updates haven't resolved this.  Let me rephrase the problems I think we have currently:

1. By silently using session ttl rather than assume role ttl for some cases we're introducing inconsistent behavior depending on the details of what's in the aws config, which seems like a bad thing.
2. The current behavior is to cache the SAML-assumed credentials for the duration of the session TTL and simply return those cached credentials, which means the actual lifetime of the credentials being returned in this case is non-deterministic and may be much shorter than expected.

I think that having a separate session TTL makes sense as we can make it much longer than the AWS-imposed 1hr max TTL for chained role assumption.  We probably just need some documentation updates to clarify how they're both used.

The downside to the solution I previously proposed is that for cases where source profile = target profile you'd likely have to re-auth/MFA to Okta every time you make a call for those credentials which also probably isn't ideal.  

At this point I'd say if source profile == target profile we should check if the cached creds expiration is > assume_role_ttl and if so just return those, and if not then refresh the cached creds and return those (would require a re-auth/MFA to Okta).  This would ensure that we're always returning credentials with a duration at least assume role ttl and minimize the amount of MFA required.
